src/main/java/space/obminyashka/items_exchange/service/impl/ImageServiceImpl.java

```diff
@@ -152,26 +143,47 @@ private Set<String> findUnsupportedType(List<MultipartFile> images) {
                 .collect(Collectors.toSet());
     }
 
+    @SneakyThrows(UnsupportedMediaTypeException.class)
+    @Override
+    public byte[] scale(MultipartFile image) {
+        validateImagesTypes(List.of(image));
+
+        try (var bais = new ByteArrayInputStream(image.getBytes());
+             var baos = new ByteArrayOutputStream()) {
+
+            return returnScaledBytes(image.getBytes(), bais, baos);
+        } catch (IOException e) {
+            log.error("An error occurred while scale an image", e);
+            throw new UnsupportedMediaTypeException(getMessageSource(
+                    ResponseMessagesHandler.ValidationMessage.INCORRECT_IMAGE_FORMAT));
+        }
+    }
+
     @Override
     public byte[] scale(byte[] bytes) {
         try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
              ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
 
-            BufferedImage originImage = ImageIO.read(bais);
-            if (originImage != null) {
-                Dimension newSize = calculatePreferThumbnailSize(
-                        new Dimension(originImage.getWidth(), originImage.getHeight()));
-                BufferedImage scaledImage = getScaled(newSize, originImage);
-                String type = URLConnection.guessContentTypeFromStream(new ByteArrayInputStream(bytes)).replace("image/", "");
-                ImageIO.write(scaledImage, type, baos);
-            }
-            return baos.toByteArray();
+            return returnScaledBytes(bytes, bais, baos);
```

Comment:
You can do vice versa:
1. Keep all the logic of scaling an image in the current method
2. Invoke the current one from **scale(MultipartFile image)** by simple passing byte array in it